pkg/kubelet/lifecycle/handlers_test.go

```diff
@@ -25,54 +25,52 @@ import (
 	"testing"
 	"time"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	containertesting "k8s.io/kubernetes/pkg/kubelet/container/testing"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
 	"k8s.io/kubernetes/pkg/kubelet/util/format"
 )
 
-func TestResolvePortInt(t *testing.T) {
-	expected := 80
-	port, err := resolvePort(intstr.FromInt(expected), &v1.Container{})
-	if port != expected {
-		t.Errorf("expected: %d, saw: %d", expected, port)
-	}
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
-}
-
-func TestResolvePortString(t *testing.T) {
-	expected := 80
-	name := "foo"
-	container := &v1.Container{
-		Ports: []v1.ContainerPort{
-			{Name: name, ContainerPort: int32(expected)},
+func TestResolvePort(t *testing.T) {
+	for _, testCase := range []struct {
+		container *v1.Container
+		name      string
+		expected  int
+	}{
+		{
+			name: "foo",
+			container: &v1.Container{
```

Comment:
I wonder if we should make just the `ContainerPort` object(s) a test case field (not the full `Container`)

Reply:
`resolvePort()` takes a `*v1.Container{}` and one of the test cases tests tests that port 80 should be returned if the container passed has no port specified (see line 50), it would be kinda hacky to have to account for that just by using `ContainerPort` instead.